authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -118,27 +128,319 @@ public List<String> policyResourceDefinesRule() {
+          .forEach(
+              table -> {
+                NameIdentifier identifier =
+                    NameIdentifier.of(authzMetadataObject.name(), table.name());
+                List<String> tabLocations =
+                    AuthorizationUtils.getMetadataObjectLocation(
+                        identifier, Entity.EntityType.TABLE);
+                tabLocations.stream()
+                    .forEach(
+                        locationPath -> {
+                          AuthorizationMetadataObject tableMetadataObject =
+                              new PathBasedMetadataObject(
+                                  authzMetadataObject.name(), table.name(), locationPath, PATH);
+                          doRemoveTableMetadataObject(tableMetadataObject);
+                        });
+                // Remove schema
+                Schema schema =
+                    GravitinoEnv.getInstance()
+                        .schemaDispatcher()
+                        .loadSchema(NameIdentifier.of(authzMetadataObject.name()));
+                List<String> schemaLocations =
+                    AuthorizationUtils.getMetadataObjectLocation(
+                        identifier, Entity.EntityType.SCHEMA);
+                schemaLocations.stream()
+                    .forEach(
+                        locationPath -> {
+                          AuthorizationMetadataObject schemaMetadataObject =
+                              new PathBasedMetadataObject(
+                                  authzMetadataObject.name(), schema.name(), locationPath, PATH);
+                          removePolicyByMetadataObject(schemaMetadataObject);
+                        });
+              });
+    }
+  }
+
+  /**
+   * Remove the TABLE, Need to remove these the relevant policies, `*.{table}`, `*.{table}.{column}`
+   * permissions.
+   */
+  private void doRemoveTableMetadataObject(AuthorizationMetadataObject authzMetadataObject) {
+    Preconditions.checkArgument(
+        authzMetadataObject instanceof PathBasedMetadataObject,
+        "The metadata object must be a PathBasedMetadataObject");
+    Preconditions.checkArgument(
+        authzMetadataObject.names().size() == 3, "The metadata object names must be 3");
+    Preconditions.checkArgument(
+        authzMetadataObject.type() == PATH, "The metadata object type must be PATH");
+    removePolicyByMetadataObject(authzMetadataObject);
+  }
+
   @Override
   protected RangerPolicy createPolicyAddResources(AuthorizationMetadataObject metadataObject) {
+    Preconditions.checkArgument(
+        metadataObject instanceof PathBasedMetadataObject,
+        "The metadata object must be a PathBasedMetadataObject");
+    PathBasedMetadataObject pathBasedMetadataObject = (PathBasedMetadataObject) metadataObject;
     RangerPolicy policy = new RangerPolicy();
     policy.setService(rangerServiceName);
-    policy.setName(metadataObject.fullName());
+    policy.setName(getAuthorizationPath(pathBasedMetadataObject));
```

Comment:
Why do we use path as the name of policy?
Maybe many file sets use the same path. What will it happen if we change the file set,?

Reply:
No, Gravitino maintained Ranger Policy only has one unique resource.